### DISCUSSION TITLE
Relative source binding in C# markup

### DISCUSSION THREAD

**Discussion by LITTOMA** [unoplatform/uno.extensions#2092] [Q&A]
How do I bind a property of a child item in a ItemsControl to a property of the parent's DataContext?

**Comment by LITTOMA** [comment]
Alright, it seems that this is not possible then. I'll find another way out.

**Comment by dansiegel** [comment]
@LITTOMA sorry I missed this before. The short answer is that in the current release of C# Markup there is no exposed property for the RelativeSource. That said I've added it to the backlog and will try to get to it this coming week. There isn't a reason it's not there other than it was just something that I didn't think to add. 

In the mean time you could try using ElementName for your binding like:

```cs
new ItemsControl()
    .Name("ItemsControl")
    .ItemTemplate<MyDataType>(dataType => new StackPanel()
        .Children(
            new Button().Content("Press Me")
                .Command(x => x.Bind("DataContext.MyCommand").ElementName("ItemsControl"))
        )
    )
```

**Reply by LITTOMA** [reply]
Thanks for the idea. This approach should works for me but my requirements could be a bit more complicated. I need a strong typed reference so that I can write some inline converters based on the data context. 

Codes should more clearly than texts. Check below pls:

``` csharp
new ItemsControl()
    .Name("ItemsControl")
    .ItemTemplate<MyDataType>(dataType => new StackPanel()
        .Children(
            new RadioButton().Content("Choose Me")
                .IsChecked(x => x.Bind("DataContext.SelectedItem").ElementName("ItemsControl").Covert(selectedItem => selectedItem == radioButton.DataContext))
        )
    )
```

**Reply by dansiegel** [reply]
Thanks for the feedback. Definitely can see some use cases for maintaining strong typing... I'm floating some ideas by the team in regard to this. We will discuss the best options on this and see what we can to improve the API in this area.
